tutorials/pubsub-spring-dedup-messages/index.md

```diff
@@ -0,0 +1,263 @@
+   > For more information, see
+   > [Granting roles to service accounts].
+
+   For more information, see
+   [Creating and managing service accounts].
+
+1. Set your `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to your service account key file.
+
+   ```shell script
+   export GOOGLE_APPLICATION_CREDENTIALS=path/to/your/credentials.json
+   ```
+   If you have not done it, set your `PROJECT_NAME` environment variable to your GCP project.
+   ```shell script
+    export PROJECT_NAME=$(gcloud config get-value project)
+   ```
+
+1. Clone this repository and navigate to the sample code:
+
+   ```shell script
+   git clone https://github.com.GoogleCloudPlatform/pubsub-spring-dedup-messages.git
+   cd community/tutorials/pubsub-spring-dedup-messages
+   ```
+
+## Bind Cloud Pub/Sub to Your Spring Boot Application
+
+Spring makes use of Spring Cloud Stream binders to send data to a Cloud Pub/Sub topic and receive data from a Cloud Pub/Sub subscription.
+
+Your application will know how to send data from an internal queue to a Cloud Pub/Sub topic if you specify 1). a Spring Cloud Stream source and 2). a Cloud Pub/Sub topic.
+
+### Specify a Spring Cloud Stream source
+Spring can recognize a Spring Cloud Stream source as a Supplier bean.
+
+In [App.java](pubsub-spring/src/main/java/com/google/example/App.java):
+
+[embedmd]:# (pubsub-spring/src/main/java/com/google/example/App.java java /  \/\/ The Supplier Bean/ /}/)
+```java
+  // The Supplier Bean makes the function a valid Spring Cloud Stream source. It
+  // sends messages to a Cloud Pub/Sub topic configured with the binding name
+  // `sendMessagesForDeduplication-out-0` in application.properties.
+  @Bean
+  Supplier<Flux<Message<String>>> sendMessagesForDeduplication(
+    final EmitterProcessor<Message<String>> frontEndListener) {
+    return () -> frontEndListener;
+  }
+```
+
+### Specify a Cloud Pub/Sub topic for the source
+Spring can find the Cloud Pub/Sub topic that your code should publish data to when you provide a topic name for your source in `application.properties` using an output binder. Here, the output binder's binding name is `sendMessagesForDeduplication-out-0`. For more information, see [Binding and Binding Names].
+
+In [application.properties](pubsub-spring/src/main/resources/application.properties):
+
+[embedmd]:# (pubsub-spring/src/main/resources/application.properties /.*Data going/ /=topicToDataflow/)
+```properties
+# Data going to Cloud Pub/Sub from a Spring Cloud Stream source defined in the
+# Supplier bean `sendMessagesForDeduplication`. The application will
+# create the Cloud Pub/Sub topic `topicToDataflow` if it does not exist.
+spring.cloud.stream.bindings.sendMessagesForDeduplication-out-0.destination=topicToDataflow
+```
+
+Similarly, your application will know how to receive data from a Cloud Pub/Sub subscription if you specify 1). a Spring Cloud Stream sink and 2). a pair of Cloud Pub/Sub topic and subscription.
```

Comment:
I'd recommend using the list formatting here. You can emphasize that both list items are required with a phrase like: "...if you specify both of the following:"